src/cdk-experimental/ui-patterns/behaviors/list-selection/list-selection.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {signal, Signal, WritableSignal} from '@angular/core';
+import {ListNavigation, ListNavigationItem} from '../list-navigation/list-navigation';
+
+/** The required properties for selection items. */
+export interface ListSelectionItem extends ListNavigationItem {
+  /** A unique identifier for the item. */
+  id: Signal<string>;
+
+  /** Whether an item is disabled. */
+  disabled: Signal<boolean>;
+}
+
+/** The required inputs for list selection. */
+export interface ListSelectionInputs<T extends ListSelectionItem> {
+  /** The items in the list. */
+  items: Signal<T[]>;
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable: Signal<boolean>;
+
+  /** The ids of the current selected items. */
+  selectedIds: WritableSignal<string[]>;
```

Comment:
Not necessarily for this PR, but one thing we should discuss in a bit more detail is how we're managing selected IDs, since I worry that the array isn't the best data structure vs. a `Set`, but we don't have a pattern established yet for a `Set` that plays well with signals